### PR TITLE
feat(patterns/pagination): add pagination pattern (#540)

### PR DESCRIPTION
**Pull request by mohamedMok**
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [x] Yes
- [ ] No

## Does this PR introduce a [breaking change](https://mozaic.adeo.cloud/Contributing/Developers/GitConventions/#breaking-changes-)?

- [ ] Yes
- [x] No

## Describe the changes

GitHub issue number or Jira issue URL: N/A

### CODE REVIEW

**Comment by mohamedMok** [comment]
https://featpagination-dot-demo-dot-mozaic-design-system-dtrp.ey.r.appspot.com/Components/Pagination/

File: packages/styles/components/_c.pagination.scss [hunk 1]
```diff
@@ -0,0 +1,58 @@
+.mc-pagination {
+  display: flex;
+  justify-content: center;
```
Comment (tiloyi) [inline packages/styles/components/_c.pagination.scss:3]: I think it's better to do this:   

```CSS
display: inline-flex;
justify-content: space-between;
```

File: packages/styles/components/_c.pagination.scss [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  &__button {
+    @include unstyle-button();
+    @include set-button-theme(get-token(color, button, 'solid-neutral'));
+    @include set-button-layout();
+
+    box-sizing: border-box;
+    min-height: $mu300;
+    min-width: $mu300;
+    margin: 0 $mu075;
```
Comment (tiloyi) [inline packages/styles/components/_c.pagination.scss:14]: ~~margin: 0 $mu075;~~
I think you can take it off the buttons and put it on the select container instead.

File: packages/styles/components/_c.pagination.scss [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        height: $mu125;
+      }
+    }
+
+    &-hidden {
+      display: none;
+    }
+  }
+
+  &__select {
```
Comment (tiloyi) [inline packages/styles/components/_c.pagination.scss:37]: ~~&__select~~ => **&__field**

```CSS
&__field {
margin: 0 $mu150;
}
```

File: packages/styles/components/_c.pagination.scss [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    margin: 0 $mu075;
+
+    &-input {
```
Comment (tiloyi) [inline packages/styles/components/_c.pagination.scss:40]: ~~&-input~~ => **&__select**

File: packages/styles/components/_c.pagination.scss [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      @include set-font-scale('05', 'm');
+
+      width: $mu900;
```
Comment (tiloyi) [inline packages/styles/components/_c.pagination.scss:43]: ~~width: $mu900;~~ => **min-width: $mu900;**

File: src/docs/Components/Pagination/code.mdx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  </a>
+</div>
+```
+
+<Preview path="pagination-default" />
+
+## Light Pagination 
+
+```html
+<div class="mc-pagination example">
```
Comment (tiloyi) [inline src/docs/Components/Pagination/code.mdx:43]: Can you remove the "example" class and put it on a separate div and add the **mc-pagination--light** modifier?

```HTML
<div class="example">
<div class="mc-pagination mc-pagination--light">...</div>
</div>
```

File: src/docs/Components/Pagination/code.mdx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+      <option value="">Page 1</option>
+    </select>
+  </div>
+
+  <a href="#" class="mc-pagination__button" aria-label="Pagination Navigation">
+    <svg class="mc-pagination__button-icon">
+      <use xlink:href="#iconRightL" />
+    </svg>
+  </a>
+</div>
```
Comment (tiloyi) [inline src/docs/Components/Pagination/code.mdx:35]: Same comment as in the **pagination.preview.html** file.

File: src/docs/Components/Pagination/code.mdx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+```
+
+<Preview path="pagination-light" />
+
+## Accessibility and semantic
+
+<HintItem>
+  Use <code>aria-label="Pagination Navigation"</code>{' '}
```
Comment (tiloyi) [inline src/docs/Components/Pagination/code.mdx:63]: Each button must have an `aria-label` attribute indicating its function (ex: `aria-label="Previous page`)

File: src/docs/Components/Pagination/code.mdx [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+</HintItem>
+
+```html
+<a 
+  href="#" 
+  class="mc-pagination__button"
+  aria-label="Pagination Navigation">
+  <!-- Your icon html tag -->
+</a>
+```
```
Comment (tiloyi) [inline src/docs/Components/Pagination/code.mdx:73]: I don't think that part's necessary.

File: src/docs/Components/Pagination/previews/pagination-default.preview.html [hunk 10]
```diff
@@ -0,0 +1,27 @@
+<div class="mc-pagination example">
```
Comment (tiloyi) [inline src/docs/Components/Pagination/previews/pagination-default.preview.html:1]: Same comment as in the **pagination.preview.html** file and can you improve the indentation of this file.

File: src/docs/Components/Pagination/previews/pagination-light.preview.html [hunk 11]
```diff
@@ -0,0 +1,21 @@
+<div class="mc-pagination example">
```
Comment (tiloyi) [inline src/docs/Components/Pagination/previews/pagination-light.preview.html:1]: Same comment as in the **pagination.preview.html** file and can you improve the indentation of this file.

File: src/docs/Components/Pagination/previews/pagination.preview.html [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    </svg>
+  </a>
+</div>
+
+<div class="mc-pagination example">
+  <a href="#" class="mc-pagination__button" aria-label="Pagination Navigation">
+    <svg class="mc-pagination__button-icon">
+      <use xlink:href="#iconLeftL" />
+    </svg>
+  </a>
```
Comment (tiloyi) [inline src/docs/Components/Pagination/previews/pagination.preview.html:25]: I think it's better to use a button in the case of pagination:
```HTML
<button type="button" class="mc-pagination__button" aria-label="Previous page">
  <svg class="mc-pagination__button-icon">
    <use xlink:href="#iconLeftL" />
  </svg>
</button>
```

File: src/docs/Components/Pagination/previews/pagination.preview.html [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+  <a href="#" class="mc-pagination__button" aria-label="Pagination Navigation">
+    <svg class="mc-pagination__button-icon">
+      <use xlink:href="#iconRightL" />
+    </svg>
+  </a>
```
Comment (tiloyi) [inline src/docs/Components/Pagination/previews/pagination.preview.html:30]: I think it's better to use a button in the case of pagination:
```HTML
<button type="button" class="mc-pagination__button" aria-label="Next page">
  <svg class="mc-pagination__button-icon">
    <use xlink:href="#iconRightL" />
  </svg>
</button>
```

File: src/docs/Components/Pagination/previews/pagination.preview.html [hunk 14]
```diff
@@ -0,0 +1,40 @@
+<div class="mc-pagination example">
+  <a href="#" class="mc-pagination__button" aria-label="Pagination Navigation">
+    <svg class="mc-pagination__button-icon">
+      <use xlink:href="#iconLeftL" />
+    </svg>
+  </a>
+  <div class="mc-pagination__select">
+    <select class="mc-select mc-pagination__select-input">
+      <option value="">Page 1</option>
```
Comment (tiloyi) [inline src/docs/Components/Pagination/previews/pagination.preview.html:9]:
```HTML
<option value="1">Page 1 of 99</option>
<option value="2">Page 2 of 99</option>
<option value="3">Page x of 99</option>
<option value="99">Page 99 of 99</option>
```

**Pull request merged**